service/status-report/main.go

```diff
@@ -117,12 +117,12 @@ func main() {
 		klog.Exitf("Failed to report status of storageClient %v: %v", storageClient.Status.ConsumerID, err)
 	}
 
-	storageClientCopy := &v1alpha1.StorageClient{}
-	storageClient.DeepCopyInto(storageClientCopy)
-	if utils.AddAnnotation(storageClient, utils.DesiredSubscriptionChannelAnnotationKey, statusResponse.DesiredClientOperatorChannel) {
-		// patch is being used here as to not have any conflicts over storageclient cr changes as this annotation value doesn't depend on storageclient spec
-		if err := cl.Patch(ctx, storageClient, client.MergeFrom(storageClientCopy)); err != nil {
-			klog.Exitf("Failed to annotate storageclient %q: %v", storageClient.Name, err)
+	// statusResponse.DesiredClientOperatorChannel will be replaced with hash field once ocs-o/r PR get merge.
+	if statusResponse.DesiredClientOperatorChannel != storageClient.Status.ProviderState {
+		storageClient.Status.ProviderState = statusResponse.DesiredClientOperatorChannel
+		statusErr := cl.Status().Update(ctx, storageClient)
+		if statusErr != nil {
+			klog.Exitf("Failed to update StorageClient status %v: %v", storageClient.Status.ConsumerID, err)
```

Comment:
This change around the subscription notification mechanism is a different concern and should be moved to its own PR